### PR TITLE
Added ref to input, auto-focus next password input after email input

### PR DESCRIPTION
**Pull request by DrySoldier**

### CODE REVIEW

File: src/components/AuthRootView/index.tsx [hunk 1]
```diff
@@ -17,24 +17,25 @@ interface IProps {
 
 const AuthRootView: FC<IProps> = ({ backgroundTitle, children }) => (
   <SafeAreaView style={styles.root}>
-    <TouchableWithoutFeedback style={styles.root} onPress={Keyboard.dismiss} accessible={false}>
-      <View style={styles.root}>
-        <View style={styles.backgroundTextContainer}>
+    <Pressable
```
Comment (DrySoldier) [inline src/components/AuthRootView/index.tsx:20]: Pressable API is more future-proof over touchable APIs

File: src/components/Input/index.tsx [hunk 2]
```diff
@@ -12,71 +18,82 @@ interface IProps {
...
+  {
+    onChangeText,
+    value,
+    leadingIcon,
+    trailingIcon,
+    textInputProps,
+    placeholder,
+  },
+  ref
+) => {
```
Comment (DrySoldier) [inline src/components/Input/index.tsx:34]: FC converted to ForwardedRef FC to give all inputs the ability to be focused on

File: src/screens/SignupScreen/index.tsx [hunk 3]
```diff
@@ -1,85 +1,106 @@
...
+        <Text style={styles.subtitleText}>{"Sign Up"}</Text>
+        <Text style={styles.descriptionText}>
+          {"Choose how you'd like to sign up."}
+        </Text>
+        <Input
+          onChangeText={setEmail}
+          value={email}
+          textInputProps={{
+            returnKeyType: "next",
+            blurOnSubmit: false,
```
Comment (DrySoldier) [inline src/screens/SignupScreen/index.tsx:60]: I was stuck on this for a bit, as I needed an `onBlur` function to trigger while still maintaining the keyboard. `onSubmitEditing` with `blurOnSubmit` set to false was my answer. `onSubmitEditing` will trigger while `blurOnSubmit` stops the Keyboard from auto-dismissing.

Unrelated, but funnily enough, `blurOnSubmit={false}` still allows the internal `onBlur` function to trigger!

File: src/components/AuthRootView/index.tsx (hunk 1 above)
Comment (benjaminharringtonrose) [reply src/components/AuthRootView/index.tsx:20]: Awesome, I didn't know that. Thanks for this! How come?

Comment (DrySoldier) [reply src/components/AuthRootView/index.tsx:20]: Just what the official docs say:

https://reactnative.dev/docs/touchablewithoutfeedback

**Pull request merged**